riscv/src/compiler.rs

```diff
@@ -95,12 +96,163 @@ impl Architecture for RiscvArchitecture {
     }
 }
 
-pub fn machine_decls() -> Vec<&'static str> {
-    vec!["use std::binary::Binary;", "use std::shift::Shift;"]
+/// Defines which coprocessors should be used by the RISCV machine.
+/// It is important to not add unused coprocessors since they may
+/// lead to many extra columns in PIL.
+#[derive(Clone, Debug, Default)]
+pub struct CoProcessors {
```

Comment:
Co-processors can and should be field-specific. I think this would be the right place to do it, right? Then the user would only specify the machines it needs but the `CoProcessor` type knows which implementation to chose. Right now, I think we should assert that the `PoseidonGL` and `GoldilocksGL` machines are only instantiated on the Goldilocks field, no?

One way to do this is to add a type parameter to `CoProcessors<T>`. I prepared this in #635.